src/appliance-onboarding-script/appliance_setup/run.py

```diff
@@ -129,17 +130,25 @@ def _populate_default_values_of_optional_fields_in_config(config):
         logging.info("avs enabled")
         avs_config_validator = ConfigValidator(config)
         avs_config_validator.validate_avs_config()
+
+        version = avs_config_validator.get_config_version()
+      
+        if(version == Constant.NEW_CONFIG_VERSION):
+            network_orchestrator: Orchestrator = NetworkOrchestrator()
+            network_orchestrator.orchestrate(config)
+
         avs_orchestrator: Orchestrator = AVSOrchestrator()
         _customer_details = avs_orchestrator.orchestrate(config)
 
+        avs_config_validator.validate_segment_details_config(_customer_details.customer_resource)
```

Comment:
>validate_segment_details_config [](http://example.com/codeflow?start=29&length=31)

Why are we validating this separately and not as a part of validate_avs_config() itself?

Reply:
I will validate in validate_avs_config() itself.

Reply:
changed the logic to avs_validate().